DashKit/DashPumpManager.swift

```diff
@@ -535,6 +535,9 @@ public class DashPumpManager: PumpManager {
                 switch result {
                 case .success(let podStatus):
                     self.mutateState({ (state) in
+                        if let finishedBolus = state.unfinalizedBolus {
+                            state.finishedDoses.append(finishedBolus)
+                        }
```

Comment:
not sure why an `unfinalizedBolus` is assigned to a `finishedBolus`?

Reply:
A dose can be finished, but not finalized (i.e. not finished with the events that process it to an unmutable state in persistent storage).